browser/src/control/VRuler.ts

```diff
@@ -0,0 +1,462 @@
+/* -*- js-indent-level: 8; fill-column: 100 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * Ruler Handler
+ */
+
+/* global L _ */
+
+interface Params {
+	[key: string]: {
+		type: string;
+		value: any;
+	};
+}
+
+interface Options {
+	interactive: boolean;
+	marginSet: boolean;
+	displayNumber: boolean;
+	tileMargin: number;
+	margin1: number | null;
+	margin2: number | null;
+	leftOffset: number | null;
+	pageOffset: number | null;
+	pageWidth: number | null;
+	pageTopMargin: number | null;
+	pageBottomMargin: number | null;
+	tabs: any[];
+	unit: string | null;
+	DraggableConvertRatio: number | null;
+	timer: ReturnType<typeof setTimeout>;
+	showruler: boolean;
+	position: string;
+	disableMarker: boolean;
+}
+
+class VRuler{
+
+	_pVerticalStartMarker: HTMLDivElement;
+	_pVerticalEndMarker: HTMLDivElement;
+	_rFace: HTMLDivElement;
+	_markerHorizontalLine: HTMLDivElement;
+	_rWrapper: HTMLDivElement;
+	_rMarginWrapper: HTMLDivElement;
+	_rBPWrapper: HTMLDivElement;
+	_rBPContainer: HTMLDivElement;
+	_rTSContainer: HTMLDivElement;
+	_tMarginMarker: HTMLDivElement;
+	_bMarginMarker: HTMLDivElement;
+	_tMarginDrag: HTMLDivElement;
+	_bMarginDrag: HTMLDivElement;
+
+	_indentationElementId: string;
+	_initialposition:number;
+	_lastposition:number;
+	_map: ReturnType<typeof L.map>;
+
+	options: Options = {
+			interactive: true,
+			marginSet: false,
+			displayNumber: true,
+			tileMargin: 20, // No idea what this means and where it comes from
+			margin1: null,
+			margin2: null,
+			leftOffset: null,
+			pageOffset: null,
+			pageWidth: null,
+			pageTopMargin: null,
+			pageBottomMargin: null,
+			tabs: [],
+			unit: null,
+			DraggableConvertRatio: null,
+			timer: null,
+			showruler: true,
+			position:'topleft',
+			disableMarker: false
+		};
+
+		constructor(map: ReturnType<typeof L.map>, options: Options) {
```

Comment:
did you run: `make prettier-write` to improve formatting?

Reply:
no, i did not run that. i just do select all then format code by right click in VS Code